### PR TITLE
[Hub Generated] Publish private branch 'EdgeMarketPlaceToMaster'

### PR DESCRIPTION
**Pull request by aneesh-ponneth**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2FAzure%2Fazure-rest-api-specs-pr%2FEdgeMarketPlaceToMaster...main/).
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by aneesh-ponneth** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/12991](https://github.com/Azure/azure-rest-api-specs-pr/pull/12991)

**Comment by v-xuto** [comment]
@aneesh-ponneth  Please contact the `ARM team` to review this PR. If you don't know how to contact `ARM team`, you can check the third of the `ARM API Review Checklist` in the [first comment](https://github.com/Azure/azure-rest-api-specs/pull/24449#issue-1757681451).

**Comment by aneesh-ponneth** [comment]
> @aneesh-ponneth Please contact the `ARM team` to review this PR. If you don't know how to contact `ARM team`, you can check the third of the `ARM API Review Checklist` in the [first comment](https://github.com/Azure/azure-rest-api-specs/pull/24449#issue-1757681451).

Hi @v-xuto , this change is already approved by ARM in this PR:  https://github.com/Azure/azure-rest-api-specs-pr/pull/12991. Do we need another ARM approval?

**Review by v-xuto — APPROVED** [review]

File: specification/edgemarketplace/resource-manager/readme.typescript.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## TypeScript
+
+These settings apply only when `--typescript` is specified on the command line.
+Please also specify `--typescript-sdks-folder=<path to root folder of your azure-sdk-for-js clone>`.
+
+```yaml $(typescript)
+typescript:
+  azure-arm: true
+  package-name: "edgemarketplace"
```
Comment (qiaozha) [inline specification/edgemarketplace/resource-manager/readme.typescript.md:9]: pleae use @azure/arm- as the package prefix,  and review the name with SDK arch board.

File: specification/edgemarketplace/resource-manager/readme.typescript.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  output-folder: "$(typescript-sdks-folder)/packages/edgemarketplace"
```
Comment (qiaozha) [inline specification/edgemarketplace/resource-manager/readme.typescript.md:10]:
```suggestion
  output-folder: "$(typescript-sdks-folder)/sdk/edgemarketplace/arm-edgemarketplace"
```
This is where we put the generated SDK in JS.

**Comment by v-xuto** [comment]
@raych1, @msyyc, @live1206 Please help review these readme md files in this PR.

File: specification/edgemarketplace/resource-manager/readme.typescript.md (hunk 1 above)
Comment (aneesh-ponneth) [reply specification/edgemarketplace/resource-manager/readme.typescript.md]: @qiaozha , Updated name. We are planning SDK release after a few months. Will do the review with SDK board.

**Comment by v-xuto** [comment]
@raych1, @msyyc, @live1206 Please help review these readme md files in this PR.

File: specification/edgemarketplace/resource-manager/readme.python.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  package-version: 2023-04-01-preview
+  clear-output-folder: true
+```
+
+### Python multi-api
+
+Generate all API versions currently shipped for this package
+
+```yaml $(python) && $(multiapi)
+batch:
```
Comment (msyyc) [inline specification/edgemarketplace/resource-manager/readme.python.md:22]: please follow https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/samplefiles/readme.python.md to update `readme.python.md`

Comment (aneesh-ponneth) [reply specification/edgemarketplace/resource-manager/readme.python.md]: fixed

**Comment by aneesh-ponneth** [comment]
@v-xuto , Can you please help understand the validation failures?
1. SDK failures - the same PR is successful in private repo. Why it is failing in public repo?
2. API readiness - The API is still in development

**Comment by v-xuto** [comment]
> @v-xuto , Can you please help understand the validation failures?
> 
> 1. SDK failures - the same PR is successful in private repo. Why it is failing in public repo?
> 2. API readiness - The API is still in development

1. There is no SDK failures, what are you referring to?
2. We can ignore the API readiness error.

**Comment by v-xuto** [comment]
@raych1,  @live1206 Please help review these readme md files in this PR.

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
Approved Go readme file.

File: specification/edgemarketplace/resource-manager/readme.csharp.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.
+
+```yaml $(csharp)
+csharp:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  payload-flattening-threshold: 1
+  clear-output-folder: true
+  client-side-validation: false
+  namespace: Microsoft.AzureStackHCI
```
Comment (live1206) [inline specification/edgemarketplace/resource-manager/readme.csharp.md:13]: What's the approved namespace for `edgemarketplace`?

File: specification/edgemarketplace/resource-manager/readme.csharp.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  output-folder: $(csharp-sdks-folder)/azurestackhci/management/Microsoft.AzureStackHCI/GeneratedProtocol
```
Comment (live1206) [inline specification/edgemarketplace/resource-manager/readme.csharp.md:14]: The output path should corresponds to `edgemarketplace`.

Comment (aneesh-ponneth) [reply specification/edgemarketplace/resource-manager/readme.csharp.md]: fixed

File: specification/edgemarketplace/resource-manager/readme.csharp.md (hunk 4 above)
Comment (aneesh-ponneth) [reply specification/edgemarketplace/resource-manager/readme.csharp.md]: fixed the namespace

**Comment by aneesh-ponneth** [comment]
> > @v-xuto , Can you please help understand the validation failures?
> > 
> > 1. SDK failures - the same PR is successful in private repo. Why it is failing in public repo?
> > 2. API readiness - The API is still in development
> 
> @aneesh-ponneth
> 
> 1. There is no SDK failures, what are you referring to?
> 2. We can ignore the API readiness error.

1-> I am referring to PS SDK failure in the validatin tools

**Comment by v-xuto** [comment]
@aneesh-ponneth Do you mean `azure-powershell` error? If yes, we can ignore it, it's not caused by this PR.

**Review by live1206 — APPROVED** [review]

**Comment by aneesh-ponneth** [comment]
> @aneesh-ponneth Do you mean `azure-powershell` error? If yes, we can ignore it, it's not caused by this PR.

@v-xuto Thank you. Since it is approved, can you please merge it?

**Comment by v-xuto** [comment]
> > @aneesh-ponneth Do you mean `azure-powershell` error? If yes, we can ignore it, it's not caused by this PR.
> 
> @v-xuto Thank you. Since it is approved, can you please merge it?

@aneesh-ponneth When do you plan to rollout this manifest?

**Comment by aneesh-ponneth** [comment]
> > > @aneesh-ponneth Do you mean `azure-powershell` error? If yes, we can ignore it, it's not caused by this PR.
> > 
> > 
> > @v-xuto Thank you. Since it is approved, can you please merge it?
> 
> @aneesh-ponneth When do you plan to rollout this manifest?

We are planning to release it next week beginning

**Comment by aneesh-ponneth** [comment]
> > > > @aneesh-ponneth Do you mean `azure-powershell` error? If yes, we can ignore it, it's not caused by this PR.
> > > 
> > > 
> > > @v-xuto Thank you. Since it is approved, can you please merge it?
> > 
> > 
> > @aneesh-ponneth When do you plan to rollout this manifest?
> 
> We are planning to release it next week beginning

May be we can publish once manifest is published. I will let you know once it is done

**Comment by aneesh-ponneth** [comment]
Can we please merge this?

**Pull request closed without merging**